### PR TITLE
Update nextalign to include OS X x86 support

### PR DESCRIPTION
**Pull request by huddlej**
Adds source URL for nextalign's prebuilt OS X x86 binary, using [preprocessing selectors](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#preprocessing-selectors). Removes platform-specific binary names from the build script and instead relies on the fact that only a single binary file with the same prefix will be downloaded for any platform (i.e., uses globbing). This approach is not ideal, but since our filenames vary by platform and [the `fn` key is not allowed in Bioconda recipes](https://bioconda.github.io/contributor/linting.html#should-not-use-fn), it seemed like an acceptable alternative.

This builds properly with the local Circle CI test on my Mac. If it builds properly here, too, I will push a commit adding support for the arm64 binary, too. 

cc: @ivan-aksamentov

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by bgruening** [comment]
@huddlej is there no way to compile from source?

**Comment by huddlej** [comment]
@bgruening, we can definitely build from source although we are a small dev team with limited experience in this kind of build management. @ivan-aksamentov is our resident expert and I think would need some extra time and perhaps assistance getting a build from source setup.

In the meantime, we are hoping to release this tool as soon as possible to our Mac OS X users who depend on our software for SARS-CoV-2 genomic analyses through [nextstrain.org](https://nextstrain.org). The number of SARS-CoV-2 sequences has become a huge bottleneck in analyses and this tool allows people to more rapidly analyze these data.

With this PR, I was hoping that [the pattern already approved and merged from an earlier PR for Linux users](https://github.com/bioconda/bioconda-recipes/pull/26910) by @cjw85 would be acceptable. Would it be possible to merge this PR as it is, so we can make the software available as soon as possible and follow up with a PR to build this package from source when we have figured out these technical details?

File: recipes/nextalign/meta.yaml [hunk 1]
```diff
@@ -6,11 +6,13 @@ package:
...
-  sha256: ea683139194d4041d52edc92a9653dcf77fa6c106281db0031bde6025a479f37
+  - url: https://github.com/nextstrain/nextclade/releases/download/nextalign-{{ version }}/nextalign-Linux-x86_64 # [linux64]
+    sha256: ea683139194d4041d52edc92a9653dcf77fa6c106281db0031bde6025a479f37                                      # [linux64]
+  - url: https://github.com/nextstrain/nextclade/releases/download/nextalign-{{ version }}/nextalign-MacOS-x86_64 # [osx]
+    sha256: f02adff831578e5297c5a8e05dbc682c335a31e6bd6c21077493aa3c1fa33ca9                                      # [osx]
 
 build:
-  number: 0
+  number: 1
   skip: True # [(not (linux and x86_64)) or osx]
```
Comment (cjw85) [inline recipes/nextalign/meta.yaml:16]: This needs amending to no longer skip osx

Comment (huddlej) [reply recipes/nextalign/meta.yaml]: Thank you for catching this, @cjw85! I just pushed a fix for this and added support for the arm64 platform.

**Comment by huddlej** [comment]
@bgruening, is there any chance we can merge this PR as is and add builds from source in the next release?

**Review by cjw85 — APPROVED** [review]

**Comment by cjw85** [comment]
@bioconda-bot please merge

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
@huddlej yes this is fine.

**Pull request merged**